### PR TITLE
chore: max_slots_per_pod can be per resource pool [DET-9771]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Put ```max_slots_per_pod``` into ```task_container_defaults```. Also mirror ```task_container_defaults``` config with rp config. 

## Test Plan

Unit test config stuff

Run a cluster with 2 GPUs on a single node.

Throw this in Helm values.
```
resourcePools:
  - pool_name: default
    task_container_defaults:
      kubernetes:
        max_slots_per_pod: 2
  - pool_name: second
```

run with global ```maxSlotsPerPod = 1``` (just by setting the helm value)

Spot check helm is set right
```
det -u admin master config | grep max_slots
  max_slots_per_pod: 1
      max_slots_per_pod: 2
    max_slots_per_pod: 1
```

Run
```
det e create examples/tutorials/mnist_pytorch/const.yaml examples/tutorials/mnist_pytorch/ --config="resources.slots_per_trial=2" --config="resources.resource_pool=default" -f
```

```kubectl get pods``` should show one pod

Run
```
det e create examples/tutorials/mnist_pytorch/const.yaml examples/tutorials/mnist_pytorch/ --config="resources.slots_per_trial=2" --config="resources.resource_pool=second" -f
```

```kubectl get pods``` should show one pod

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/pkg/model/task_container_defaults.go [hunk 1]
```diff
@@ -87,6 +92,12 @@ func (c *TaskContainerDefaultsConfig) Validate() []error {
 	return errs
 }
 
+// KubernetesTaskContainerDefaults is task container defaults specific to Kubernetes.
+// TODO eventually move podSpec here and eventually move this to expconf too.
```
Comment (NicholasBlaskey) [inline master/pkg/model/task_container_defaults.go:96]: I'll fill in JIRA and make tickets for these after review making sure this reasonable

File: docs/reference/deploy/config/master-config-reference.rst [hunk 2]
```diff
@@ -151,6 +151,12 @@ The default list of devices to pass to the Docker daemon. Ignored by resource ma
 The default bind mounts to pass to the Docker container. Ignored by resource managers of type
 ``kubernetes``. See :ref:`resources.devices <exp-bind-mounts>` for more details.
 
+``kubernetes``
+==============
+
+-  ``max_slots_per_pod`` See :ref:`resource_manager.max_slots
+   <master-config-reference-max-slots-per-pod>` for more details.
+
```
Comment (tara-det-ai) [inline docs/reference/deploy/config/master-config-reference.rst:159]: does not need to be a bullet, that would make it different from the rest of the page

``kubernetes``
==============

``max_slots_per_pod`` See :ref:`resource_manager.max_slots
   <master-config-reference-max-slots-per-pod>` for more details.

**Review by tara-det-ai — COMMENTED** [review]
remove the bullet formatting
https://github.com/determined-ai/determined/pull/7923/files?file-filters%5B%5D=.go&file-filters%5B%5D=.rst&file-filters%5B%5D=.yaml&owned-by%5B%5D=tara-det-ai&show-viewed-files=true#r1331804640

File: master/pkg/model/task_container_defaults.go (hunk 1 above)
Comment (hamidzr) [reply master/pkg/model/task_container_defaults.go:96]: makes sense to me.

**Pull request merged**